app/models/prisoner_step.rb

```diff
@@ -16,6 +16,7 @@ class PrisonerStep
   delegate :name, to: :prison, prefix: true
 
   def prison
-    Prison.find_by(id: prison_id)
+    # Memoize to avoid multiple API lookups
+    @_prison ||= prison_id ? Prison.find_by_id(prison_id) : nil
```

Comment:
Right, I'm definitely renaming this method – you and Alan have both been fooled into thinking that Prison is an ActiveRecord object (it is not, it's doing an API lookup).